compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRTypeWriter.java

```diff
@@ -343,9 +344,17 @@ private void writeObjectAndServiceTypes(BObjectType bObjectType) {
             attachedFuncs = new ArrayList<>();
             buff.writeByte(0); // constructor not present
         }
-        buff.writeInt(attachedFuncs.size());
+
+        List<BAttachedFunction> writableAttachedFunctions = new ArrayList<>();
         for (BAttachedFunction attachedFunc : attachedFuncs) {
-            writeAttachFunction(attachedFunc);
+            if (bObjectType.getKind() == TypeKind.SERVICE && attachedFunc.funcName == Names.DEFAULT_INIT_SUFFIX) {
```

Comment:
shall we move this check to runtime?  as we have discussed, the proper fix would be to keep these as separate functions in the type, but for the time being +1 for moving this check to runtime

Reply:
This is changed in 94dc802.